libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -470,6 +470,35 @@ func (segments *diskQueueSegments) sizeOnDisk() uint64 {
 	return total
 }
 
+// Iterates through all segment types to find the oldest ID in the queue
+func (segments *diskQueueSegments) oldestIDOnDisk() segmentID {
+	// not all segment types are pre-sorted, hence us appending some, taking [0] from others
+	oldSegments := []*queueSegment{}
+
+	if len(segments.reading) > 0 {
+		oldSegments = append(oldSegments, segments.reading[0])
+	}
+	if len(segments.acking) > 0 {
+		oldSegments = append(oldSegments, segments.acking[0])
+	}
+
+	oldSegments = append(oldSegments, segments.writing...)
+	oldSegments = append(oldSegments, segments.acked...)
+
+	sort.Slice(oldSegments, func(i, j int) bool { return oldSegments[i].id < oldSegments[j].id })
```

Comment:
Is this taking the oldest segment ID, or the oldest event ID? I think it is the former, and we want the latter.

We report the oldest event ID in the memory queue, which has no concept of a segment:

https://github.com/elastic/beats/blob/df5c62f96a8483fe809d01011a28979ec9d44437/libbeat/publisher/queue/memqueue/eventloop.go#L155-L161

Probably implementing this requires https://github.com/elastic/elastic-agent-shipper/issues/27 to be done first

Reply:
AH, you're right. @leehinman Do you think there's any value in reporting the oldest segment ID? The more I think about it, I'm kind of leaning towards "no."

Reply:
No, I don't think oldest ID will be useful.  Also it is easy to find out, the name of segment file is `<id no>.seg`, so just looking at the queue directory gives us this info.